api/src/main/java/com/google/sps/SendNotifications.java

```diff
@@ -1,117 +1,155 @@
+        String playerEmail = usersDocSnap.getString("email");
+        String hostName = players.document(playerNames.get(0)).get().get().getString("username");
 
         // Adding player to Email object
-        emails.add(new Email(new User(playerEmail, playerName)));
-
-        // Adds Start game message to the email object
-        if(emails.isEmpty()) {
+        emails.add(emailType.createEmail(gameID, playerEmail, hostName));
+        
+        // Checks if there are any players
+        if (emails.isEmpty()) {
           throw new IllegalArgumentException();
-        } else {
-          emails.get(emails.size()-1).playerTurn(gameID);
         }
       }
     } catch (Exception e) {
-      System.out.println(e);
+      System.out.println("Exception in gatherRecipients" + e);
     }
 
     return emails;
   }
 
-  @GetMapping("/notifyTurn")
-  public void playerTurnEmail() throws IOException {
-    List<Email> emails = gatherRecipients();
+  public EmailCreation getNextPlayer(HttpServletRequest request, HttpServletResponse response)
+    throws IOException {
+
+    Firestore db = FirestoreClient.getFirestore();
+    String gameID = request.getParameter("gameID");
+    DocumentReference game = db.collection("games").document(gameID);
+    CollectionReference players = db.collection("users");
+    EmailCreation notification = null;
+
+    try {
+      DocumentSnapshot gameDocSnap = game.get().get();
+      int currentPlayerIndex = (int) ((long) gameDocSnap.get("currentPlayerIndex"));
+      String playerName = ((List<String>) gameDocSnap.get("players")).get(currentPlayerIndex);
+
+      DocumentSnapshot usersDocSnap = players.document(playerName).get().get();
+
+      // Retrieves player information
+      String playerEmail = usersDocSnap.getString("email");
+
+      notification = EmailCreation.playerTurnEmail(gameID, new User(playerEmail, playerName));
+    } catch (Exception e) {
+      System.out.println("Exception with DocumentSnapshot" + e);
+    }
+
+    if (notification == null) {
+      throw new IllegalArgumentException();
+    }
+
+    return notification;
+  }
+
+  @PostMapping("/api/notify")
+  public void sendEmail(HttpServletRequest request, HttpServletResponse response)
+      throws IOException, InterruptedException, ExecutionException {
+
+    Firebase.init();
+    Firestore db = FirestoreClient.getFirestore();
+
+    EmailType emailType = EmailType.valueOf(request.getParameter("emailType").toUpperCase());
+    String gameID = request.getParameter("gameID");
     final String FROM = "pictophone.noreply@gmail.com";
 
-    try{
-      Gmail service = ServiceCreation.createService();
+    DocumentSnapshot docSnap = db.collection("games").document(gameID).get().get();
+    int amtOfPlayers = ((List<String>) docSnap.get("players")).size();
+    int currentPlayer = (int) ((long) docSnap.get("currentPlayerIndex"));
+
+    if (emailType == EmailType.START || (currentPlayer+1) > amtOfPlayers) {
```

Comment:
what's the reasoning behind not having an END email type (and instead manually calculating like this), but there being an email type for both START and TURN?

Reply:
the manual calculation would need to happen somewhere and at the moment, the client wasn't sending the end type anywhere, so i decided to get rid of it

Reply:
It doesn't seem consistent for the client to send the emailType on START and TURN but not end

Reply:
well the start would be computed through the game creation button and the send function sends an email each turn. i can't think of the reason that i did it, but do you think it's better for the check to be in the front end?

Reply:
I think it's better to be consistent, one way or another

Reply:
so essentially, no matter what way we go about it, there should be an end variable to denote?

Reply:
umm not really sure what you mean by that, but i believe in you